### PR TITLE
KAFKA-10658 ErrantRecordReporter.report always return completed futur…

### PR DESCRIPTION
**Pull request by chia7712**
issue: https://issues.apache.org/jira/browse/KAFKA-10658

This issue happens when both DLQ and error log are enabled. There is a incorrect filter in handling multiple reports and it results in the uncompleted future is filtered out. Hence, users always receive a completed future even though the record is still in producer buffer.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@aakashnshah @rhauch @kkonstantine Could you take a look? This issue was related to #8720 and you guys are experts on that.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ProcessingContextTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+public class ProcessingContextTest {
+
+    @Test
+    public void testReportWithSingleReporter() {
+        testReport(1);
+    }
+
+    @Test
+    public void testReportWithTwoReporters() {
```
Comment (showuon) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ProcessingContextTest.java]: testReportWithTwoReporters -> testReportWithMultipleReporters

**Review by showuon — COMMENTED** [review]
Nice find! Left a small suggestion!

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ProcessingContextTest.java (hunk 1 above)
Comment (chia7712) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ProcessingContextTest.java]: will copy that

**Comment by chia7712** [comment]
@dajac Could you please take a look?

**Review by kkonstantine — APPROVED** [review]
Nice find. 
LGTM

Good to backport up to 2.6.

**Comment by chia7712** [comment]
merge to trunk, 2.7 and 2.6

**Pull request merged**